WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -151,10 +152,11 @@ private void addAdvancedSearchButton()
             DynamicGrid.Children.Add(advancedSearch);
             advancedSearch.Click += (s, e) =>
             {
-                MapObject chosenMapObject = row[Grid.GetRow(advancedSearch)];
+                
                 if (row.ContainsKey(Grid.GetRow(advancedSearch)))
                 {
-                    selectedObject = chosenMapObject;
+                    MapObject chosenMapObject = row[Grid.GetRow(advancedSearch)];
+                    selectedObjectId = row[Grid.GetRow(advancedSearch)].Id;
```

Comment:
DRY - it is better to use the variable you have defined. Something like "selectedObjectId = chosenMapObject.id"

Reply:
Fixed.